### PR TITLE
Add commits_detail_page, commits_contributor_list, commits_organization_page, commits_repo_page interface

### PR DESCRIPTION
**Pull request by lishengbao**

### CODE REVIEW

File: app/graphql/types/queries/base_query.rb [hunk 1]
```diff
@@ -135,6 +135,13 @@ def validate_date!(current_user, label, level, begin_date, end_date)
               )
       end
 
+      def validate_admin(current_user)
```
Comment (EdmondFrank) [inline app/graphql/types/queries/base_query.rb:138]: Methods that throw exceptions should have a `!` symbol after the method name.

File: app/graphql/types/queries/base_query.rb [hunk 2]
```diff
@@ -135,6 +135,13 @@ def validate_date!(current_user, label, level, begin_date, end_date)
...
+        login_required!(current_user)
+        _admin = current_user&.is_admin?
+        return if _admin
```
Comment (EdmondFrank) [inline app/graphql/types/queries/base_query.rb:141]: The intermediate variable `_admin` does not seem to be needed here.

File: app/graphql/types/queries/commits_detail_page_query.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        resp = indexer.fetch_commit_page_by_repo_urls(repo_urls, begin_date, end_date, branch, per: per, page: page,
+                                                      filter_opts: filter_opts, sort_opts: sort_opts)
+
+        count = indexer.commit_count_by_repo_urls(repo_urls, begin_date, end_date, branch, filter_opts: filter_opts)
+        hits = resp&.[]('hits')&.[]('hits') || []
+
+        domain_list = hits.map { |data| data.dig('_source','author_domain') }.to_set.to_a
+        domain_map = Organization.map_by_domain_list(domain_list)
+
+        commit_hash_list = hits.map { |data| data.dig('_source','hash') }.to_set.to_a
```
Comment (EdmondFrank) [inline app/graphql/types/queries/commits_detail_page_query.rb:46]: Note the format of the code: `,` in the `dig` method should be followed by a space.

File: app/models/concerns/commit_enrich.rb [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            "terms": { "field": "author_domain", "size": per },
+            "aggs": {
+              "lines_changed": { "sum": { "field": "lines_changed" } },
+              "lines_added": { "sum": { "field": "lines_added" } },
+              "lines_removed": { "sum": { "field": "lines_removed" } },
+              "sort_bucket": { "bucket_sort": { "sort": [{ "lines_changed": { "order": "desc" } }] } }
+            }
+          })
+        .per(0)
+        .execute
```
Comment (EdmondFrank) [inline app/models/concerns/commit_enrich.rb:122]: You can use the symlbol key of Hash directly, without adding `"` to the Hash key.

File: app/models/organization.rb [hunk 5]
```diff
@@ -14,4 +14,34 @@ def self.mapping
...
+             .per(org_name_list.length)
+             .execute
+             .raw_response
+    hits = resp&.[]('hits')&.[]('hits') || []
+    hits.map{ |item| item['_source']['domain'] }
+  end
+
+  def self.domain_list()
+    resp = self
+             .per(10000)
```
Comment (EdmondFrank) [inline app/models/organization.rb:40]: Use constant definitions and avoid "magic numbers".

**Review by EdmondFrank — CHANGES_REQUESTED** [review]
Please adjust the pull code according to the comments

**Comment by lishengbao** [comment]
> Please adjust the pull code according to the comments

Resubmitted as requested

**Review by EdmondFrank — COMMENTED** [review]
LGTM

**Review by EdmondFrank — APPROVED** [review]

**Review by eyehwan — APPROVED** [review]

**Comment by eyehwan** [comment]
@EdmondFrank you can help merge it.

**Pull request merged**